src/main/java/frc/robot/command/PPSwerveControllerCommand.java

```diff
@@ -0,0 +1,102 @@
+package frc.robot.command;
+
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.controller.HolonomicDriveController;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.kinematics.SwerveDriveKinematics;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import com.pathplanner.lib.controllers.PPHolonomicDriveController;
+
+/**
+ * PPSwerveControllerCommand follows a given trajectory using a holonomic drive controller.
+ * It samples the trajectory, calculates the desired chassis speeds, converts them to swerve
+ * module states using the provided kinematics, and sends the commands via outputModuleStates.
+ */
+public class PPSwerveControllerCommand extends Command {
```

Comment:
This whole class is redundant. Refer to [PathPlannerCommands#getFollowPathCommand](https://github.com/momentumfrc/2025Reefscape/blob/main/src/main/java/frc/robot/utils/PathPlannerCommands.java)